### DISCUSSION TITLE
Widget are referencing css class which does not exists in src

### DISCUSSION THREAD

**Discussion by valentinMachado** [iTowns/itowns#2063] [Q&A] [answered]
[Widget class](https://github.com/iTowns/itowns/blob/master/src/Utils/gui/Widget.js) add css class to its domElement but these class does not exists in src but in [examples/css/widgets.css](https://github.com/iTowns/itowns/blob/master/examples/css/widgets.css).

Do you consider that the user has to know that he has to define these classes on its own ?
Why not setting css in the domElement directly ? (making clearer that Widget is handling some css and working on its own)
[create css class in javascript](https://stackoverflow.com/questions/1720320/how-to-dynamically-create-css-class-in-javascript-and-apply) or [webpack css](https://webpack.js.org/loaders/css-loader/)
Why not remove all css behavior if you consider that this is the user which have to deals with it ? (making this class quite empty)

**Comment by jailln** [comment, marked as answer]
I agree with your last proposition that we should leave everything regarding style to the user, see https://github.com/iTowns/itowns/issues/2005 (the css part). You can modify it accordingly
